src/toil/jobStores/abstractJobStore.py

```diff
@@ -903,6 +904,24 @@ def writeFileStream(self, jobStoreID=None, cleanup=False, basename=None):
                file basename so that when searching the job store with a query
                matching that basename, the file will be detected.
 
+        :param str mode: an optional string that specifies the mode in which the file is opened.
+                It defaults to 'b' which means open for writing in binary mode.
+
+                Currently supported mode characters:
+
+                    - 'b': open for writing in binary mode
+
+                    - 't': open for writing in text mode
```

Comment:
Hmmm... I dislike going against conventional standards for well established use-cases and `t` isn't a standard.  I think that we should probably remove `mode='b'` and use `bytes` when `encoding=None`, otherwise, use the encoding.